Automattic/wp-calypso
# Pattern Library: Don't run `usePatterns` on home page

Fixes https://github.com/Automattic/dotcom-forge/issues/6532

## Proposed Changes

As a follow-up to #89475, this PR disables the `usePatterns` query on the home page until a search is submitted or a category is requested. 

This new behavior means that there's a brief loading time after users have submitted a search. To manage this, I've tweaked the pattern library heading to `Loading patterns…` and added a placeholder pattern.

https://github.com/Automattic/wp-calypso/assets/1101677/81c98f2d-f92b-4007-b66e-f612e80e78cc

## Testing Instructions



1. Navigate to `/patterns`
2. Submit a search
3. Ensure that you see the `Loading patterns…` heading
4. Click on a category link
5. Ensure that the pattern previews are there instantly (there is no short moment when the list of patterns is empty)
